### DISCUSSION TITLE
Why are those queries giving different results?

### DISCUSSION THREAD

**Discussion by randbytes** [prisma/prisma#22969] [Q&A] [answered]
### Question

```js
{"AND":[{"prop1":{"isNot":null}},{"prop2":{"is":{"name":{"equals":"TEST"}}}}]}
{"AND":[{"prop2":{"is":{"name":{"equals":"TEST"}}}},{"prop1":{"isNot":null}}]}
```

### How to reproduce (optional)

```js
await prisma.model.count({where: {AND: [{prop1: {isNot: null}}, {prop2: {is: {name: "TEST"}}}]}})
await prisma.model.count({where: {AND: [{prop2: {is: {name: "TEST"}}}, {prop1: {isNot: null}}]}})
```

### Expected behavior (optional)

Same result

### Information about Prisma Schema, Client Queries and Environment (optional)

generator client {
  provider = "prisma-client-js"
  binaryTargets = ["native", "linux-musl-openssl-3.0.x"]
}
datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}
@prisma/client = 5.9.0

**Comment by ludralph** [comment, marked as answer]
Hi @randbytes 👋 

Thank you for raising this question.

The two queries provided are logically equivalent and should indeed produce the same result when executed. We do have an open [issue](https://github.com/prisma/prisma/issues/22263) to address this. You can please add this example to that issue to help our engineers in their investigation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by janpio** [reply]
Note, here is also another issue that @randbytes opened: https://github.com/prisma/prisma/issues/22970
